htdocs/core/class/html.form.class.php

```diff
@@ -1666,6 +1666,15 @@ public function selectcontacts($socid, $selected = '', $htmlname = 'contactid',
 		if ($socid > 0 || $socid == -1) {
 			$sql .= " AND sp.fk_soc = ".((int) $socid);
 		}
+		// Show contact of parent
+    	if (!empty($conf->global->CONTACT_ON_DIFFERENT_THIRDPARTIES)) {
+    		require_once DOL_DOCUMENT_ROOT.'/societe/class/societe.class.php';
+			$soc = new Societe($db);
+			$result = $soc->fetch($socid);
+			if ($result > 0 && !empty($soc->parent)) {
+				$sql .= ' OR sp.fk_soc = '.$soc->parent;
```

Comment:
Surely a bug here. You add a OR be before you may have a AND and after another AND.
You must introduce parenthesis on test for fk_soc to avoid unexepected effects.
Also can you confirm that option CONTACT_ON_DIFFERENT_THIRDPARTIES means in fact
CONTACT_SHOW_ALSO_PARENT_COMPANY_CONTACTS (i mean it show also contact of parent company but no other thirdpary ?).
If yes can you use this name CONTACT_SHOW_ALSO_PARENT_COMPANY_CONTACTS

Reply:
Hi. Is right. Fixed